students/smitco/lesson03/mailroom.py

```diff
@@ -44,25 +45,43 @@ def intro():
     else:
         return addressee    
 def donation(n):
-    dollars = input("What is the donation amount? ")
-    if int(dollars) < (10 ** 6):
-        position = donors.index(n)
-        a = amounts[position]
-        a = a + (int(dollars),)
-        amounts[position] = a
-        return dollars
+    if n == "quit":
+        print()
+        pass
     else:
-        print("\nNice try, Moneybags.")
-        quit()
+        dollars = input("What is the donation amount? ")
+        if dollars == "quit":
+            donors.remove(n)
+            print()
+            pass
+        elif int(dollars) < (10 ** 6):
+            position = donors.index(n)
+            a = amounts[position]
+            a = a + (int(dollars),)
+            amounts[position] = a
+            return dollars
+        else:
+            print("\nNice try, Moneybags.\n")
+            quit()
 def card(n, v):
-    print("\nThank you, %s, for supporting The Derek Zoolander School for Kids Who Can't Read Good and Want to Do Other Stuff Good Too. Here, we teach kids that there's more to life than being really, really ridiculously good-looking. Because of your generous donation of $%s, we will be able to build a center that is not for ants but can fit real children.\n" % (n, v))
+    if v == None:
+        pass
+    else:
+        print("\nThank you, %s, for supporting The Brave Heart Foundation. Your generous donation of $%s will make a positive, life-changing impact for teens nationwide.\n" % (n, v))
 def quit():
     pass
 
-        
-        
-        
-if __name__ == '__main__':        
+def report():
+    print()
+    print("{:<20} {:<15} {:^12} {:<15}".format("Donor", "Total Given", "Number", "Average Given"))
+    for i in range(len(donors)):
+        tot = sum(amounts[i])
+        num = len(amounts[i])
+        print("{:<20} ${:^15.2f} {:^12} ${:^15.2f}".format(donors[i], tot, num, tot/num))
+    print()
```

Comment:
this report is supposed to be sorted. please review instructions and add the functionality for next time.